### PR TITLE
[SDKS-8265] GTE, LTE and Between Semver matchers

### PR DESCRIPTION
**Pull request by EmilianoSanchez**
# JavaScript commons library

## Extra Notes

Matcher exceptions are handled and logged in the `parser` module, to reuse code.

### CODE REVIEW

File: src/evaluator/matchers/semver_gte.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { ILogger } from '../../logger/types';
+import { Semver } from '../../utils/Semver';
+
+export function greaterThanEqualToSemverMatcherContext(log: ILogger, ruleAttr: string) {
+  const ruleSemver = new Semver(ruleAttr);
+
+  return function greaterThanEqualToSemverMatcher(runtimeAttr: string): boolean {
+    const runtimeSemver = new Semver(runtimeAttr);
```
Comment (sanzmauro) [inline src/evaluator/matchers/semver_gte.ts:8]: What happen is Semver class throws an exception at this point?

File: src/evaluator/matchers/semver_lte.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { ILogger } from '../../logger/types';
+import { Semver } from '../../utils/Semver';
+
+export function lessThanEqualToSemverMatcherContext(log: ILogger, ruleAttr: string) {
+  const ruleSemver = new Semver(ruleAttr);
```
Comment (sanzmauro) [inline src/evaluator/matchers/semver_lte.ts:5]: same here?

File: src/evaluator/matchers/semver_gte.ts (hunk 1 above)
Comment (EmilianoSanchez) [reply src/evaluator/matchers/semver_gte.ts:8]: Matcher exceptions are handled in the `parser` function, which calls the matchers.

It's done this way to reduce code size, by handling and logging the errors in a single place.

File: src/evaluator/matchers/semver_lte.ts (hunk 2 above)
Comment (EmilianoSanchez) [reply src/evaluator/matchers/semver_lte.ts:5]: Idem. In JS we have to minimize code size, so we try to reuse as much code as possible. 
Exception handling is in the `parser` function at `src/evaluator/parser/index.ts`.

**Review by sanzmauro — APPROVED** [review]

**Pull request merged**